### PR TITLE
fix(Snooze): Add unsnooze action

### PR DESCRIPTION
**Pull request by JohannesGGE**
Fix: #8744

Replaces the Snooze button with a Unsnooze button if the selected mailbox is the Snoozed-Mailbox and add unsnooze functionality.

+Mark message as unread after wake

### CODE REVIEW

File: appinfo/routes.php [hunk 1]
```diff
@@ -375,6 +380,11 @@
 			'url' => '/api/thread/{id}/snooze',
 			'verb' => 'POST'
 		],
+		[
+			'name' => 'thread#unSnooze',
+			'url' => '/api/thread/{id}/unsnooze',
+			'verb' => 'GET'
```
Comment (ChristophWurst) [inline appinfo/routes.php:386]: POST

File: lib/Service/SnoozeService.php [hunk 2]
```diff
@@ -151,12 +191,72 @@ public function snoozeThread(
...
+	 */
+	public function unSnoozeThread(
+		Message $selectedMessage,
+		?string $currentUserId,
+	):void {
+		$snoozedMailbox = $this->mailManager->getMailbox($currentUserId, $selectedMessage->getMailboxId());
+		$srcAccount = $this->accountService->find($currentUserId, $snoozedMailbox->getAccountId());
+
+		$mailAccount = $srcAccount->getMailAccount();
+		$messageInTrash = $snoozedMailbox->getId() === $mailAccount->getTrashMailboxId(); //Nicht unbedingt nötig?
```
Comment (ChristophWurst) [inline lib/Service/SnoozeService.php:220]: :de:

File: lib/Service/SnoozeService.php [hunk 3]
```diff
@@ -267,8 +339,6 @@ private function wakeMessagesByAccount(Account $account): void {
 					$srcMailboxName
 				);
 
-				//TODO mark message as unread?
```
Comment (ChristophWurst) [inline lib/Service/SnoozeService.php:270]: is that done?

**Review by st3iny — APPROVED** [review]
Tested and works.

Restored messages is marked as read though. Should that be fixed?

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**